docs/spec/lightclient/detection/detection.md

```diff
@@ -1,19 +1,199 @@
+    - For each secondary, we check the primary against one secondary
+
+
+- Observe that just two blocks for the same height are not
+sufficient proof of fork.
+One of the blocks may be bogus [TMBC-BOGUS.1] which does
+not constitute slashable behavior.  
+Which leads to the question whether the light node should try to do
+fork detection on its initial block (from subjective
+initialization). This could be done by doing backwards verification
+(with the hashes) until a bifurcation block is found. 
+While there are scenarios where a
+fork could be found, there is also the scenario where a faulty full
+node feeds the light node with bogus light blocks and forces the light
+node to check hashes until a bogus chain is out of the trusting period.
+As a result, the light client
+should not try to detect a fork for its initial header. **The initial
+header must be trusted as is.**
+
+
+# Light Client Sequential Supervisor
+
+**TODO:** decide where (into which specification) to put the
+following:
+
+
+We describe the context on which the fork detector is called by giving
+a sequential version of the supervisor function.
+Roughly, it alternates two phases namely:
+   - Light Client Verification. As a result, a header of the required
+     height has been downloaded from and verified with the primary
+   - Light Client Fork Detections. As a result the header has been
+     cross-checked with the secondaries. In case the is a fork we
+     submit "proof of fork" and exit.
+   
+	 
+
+
 
-- clarify EXPIRED case. Can we always punish? Can we give sufficient
-  conditions. 
+
+#### **[LC-FUNC-SUPERVISOR.1]:**
+
+```go
+func Sequential-Supervisor () (Error) {
+    loop {
+	    // get the next height
+        nextheight := input();
+		
+		// Verify
+        result := NoResult;
+        while result != ResultSuccess {
+            lightStore,result := VerifyToTarget(primary, lightStore, nextheight);
+            if result == ResultFailure {				
+				// pick new primary (promote a secondary to primary)
+				/// and delete all lightblocks above
+	            // LastTrusted (they have not been cross-checked)
+	            Replace_Primary();
+			}
+        }
+		
+		// Cross-check
+        PoFs := Forkdetector(lightStore, PoFs);
+        if PoFs.Empty {
+		    // no fork detected with secondaries, we trust the new
+			// lightblock
+            LightStore.Update(testedLB, StateTrusted);
+        } 
+        else {
+		    // there is a fork, we submit the proofs and exit
+            for i, p range PoFs {
+                SubmitProofOfFork(p);
+            } 
+            return(ErrorFork);
+        }
+    }
+}
+```
+**TODO:** finish conditions
+- Implementation remark
+- Expected precondition
+    - *lightStore* initialized with trusted header
+	- *PoFs* empty
+- Expected postcondition
+    - runs forever, or
+	- is terminated by user and satisfies LightStore invariant, or **TODO**
+	- has submitted proof of fork upon detecting a fork
```

Comment:
We might want to clarify timing expectation in case of fork on the primary: if a forked block is verified at time t, the corresponding fork is retrieved the latest at time t + Fork_Detection_Timeout.